packages/core/src/stateUtils.ts

```diff
@@ -505,7 +498,7 @@ export function formatTransitions<
       )
     : [];
   const eventlessConfig = stateNode.config.always
```

Comment:
it feels like this should be put on a separate key on the state - instead of mixing this together with other, regular, transitions

this would make the `selectEventlessTransitions` slightly easier and would make it more performant - we wouldn't have to iterate through the whole list of transitions just to get to the eventless ones. We could move `eventlessConfig` to the the beginning of the `formattedTransitions` but then we'd still have to implement a stop condition in the `selectEventlessTransitions` looping. Separating those transition types would just make it working well out of the box - and it would make reading this information in the editor and stuff easier as one would only have to access a single property to read those special transitions

Reply:
These are now on `stateNode.always`